plasmapy/plasma/tests/test_grids.py

```diff
@@ -11,190 +11,117 @@
     shape = grid.shape
     unit = grid.units
 
     # Grid should be uniform
-    assert grid.uniform_grid == True
+    assert grid.is_uniform_grid == True
 
     # Test initializing with a provided grid
-    grid2 = grids.CartesianGrid(grid.grid)
-
-    # Test initializing using the unit keyword
-    grid = grids.CartesianGrid(1, 1, units=u.cm)
-    grid2 = grids.CartesianGrid(grid.grid.value, units=grid.units)
+    grid2 = grids.CartesianGrid(grid.grids[0], grid.grids[1], grid.grids[2],)
 
     # Units not all consistent
     with pytest.raises(ValueError):
-        grid = grids.CartesianGrid(-1, 1, units=[u.m, u.rad, u.rad])
-
-    with pytest.raises(ValueError):
-        grid = grids.CartesianGrid(-1, 1, units=[u.m, u.rad, u.m])
+        grid = grids.CartesianGrid(
+            [-1 * u.m, -1 * u.rad, -1 * u.m], [1 * u.m, 1 * u.rad, 1 * u.m]
+        )
 
 
 def test_interpolators():
     # Test interpolator
 
     # Test interpolator
     # Create grid
-    grid = grids.CartesianGrid(-1 * u.cm, 1 * u.cm, num=100)
+    grid = grids.CartesianGrid(-1 * u.cm, 1 * u.cm, num=25)
+    # Add some data to the grid
+    data = grid.add_quantity("pts0", grid.grids[0])
+    data = grid.add_quantity("pts1", grid.grids[1])
+
     pos = np.array([0.1, -0.3, 0]) * u.cm
+    pos2 = np.array([[0.1, -0.3, 0], [0.1, -0.3, 0]]) * u.cm
 
     # Interpolate indices
-    i = grid.interpolate_indices(pos)
-    pout = grid.grid[i[0][0], i[0][1], i[0][2], :]
+    i = grid.interpolate_indices(pos)[0]
+
+    pout = grid.grid[i[0], i[1], i[2], :] * grid.unit
+
     # Assert that nearest grid cell was found
-    assert np.allclose(pos, pout, atol=0.03)
+    assert np.allclose(pos, pout, atol=0.1)
 
     # Interpolate grid values using nearest-neighbor interpolator
-    pout = grid.nearest_neighbor_interpolator(pos, grid.grid)
-    assert np.allclose(pos, pout, atol=0.03)
+    pout = grid.nearest_neighbor_interpolator(pos, "pts0")
+    assert np.allclose(pos[0], pout, atol=0.1)
 
     # Interpolate grid values using volume-weighted interpolator
-    pout = grid.volume_averaged_interpolator(pos, grid.grid)
-    assert np.allclose(pos, pout, atol=0.03)
+    pout = grid.volume_averaged_interpolator(pos, "pts0")
+    assert np.allclose(pos[0], pout, atol=0.1)
 
     # Test using multiple arguments
-    pout, pout2 = grid.nearest_neighbor_interpolator(pos, grid.grid, grid.grid * 2)
+    pout, pout2 = grid.nearest_neighbor_interpolator(pos, "pts0", "pts1")
 
-    # Test a u-quantity input
-    i = grid.interpolate_indices(pos.value)
-    pout = grid.nearest_neighbor_interpolator(pos.value, grid.grid)
     # Test multiple grid points
     pos = np.array([[0, 0, 0], [0.5, 0.2, 0]]) * u.cm
     i = grid.interpolate_indices(pos)
-    pout = grid.nearest_neighbor_interpolator(pos.value, grid.grid)
-    pout = grid.volume_averaged_interpolator(pos, grid.grid)
+    pout = grid.nearest_neighbor_interpolator(pos.value, "pts0")
+    pout = grid.volume_averaged_interpolator(pos, "pts0")
```

Comment:
I'm starting to think it might be best to implement those ourselves using jitclass. That way, the API could be kept similar to scipy's.

Reply:
Sure - note that both are ultimately relying on interpolate_indices right now, so the root of the problem still lies there.